blocks/library/text/index.js

```diff
@@ -31,13 +31,29 @@ registerBlockType( 'core/text', {
 		content: query( 'p', children() ),
 	},
 
+	transforms: {
+		from: [
+			{
+				type: 'raw',
+				matcher: ( node ) => (
+					node.nodeName === 'P' &&
```

Comment:
Note that this is paste, not legacy content. Fine with matching current alignment classes.